src/app/components/vm-list/vm-list.component.ts

```diff
@@ -270,10 +303,146 @@ export class VmListComponent implements OnInit, AfterViewInit {
+    There is no and operator because that is the default behavior. ie foo bar is logically equivalent to foo and bar
+    keywords/operations can be escaped with \, no need to do so inside quotes though
+  */
+  private parseSearch(search: string) {
+    let parsed = new Array<SearchTerm>();
+    const tokens = search.split(' ');
+    for (let i = 0; i < tokens.length; i++) {
+      const token = tokens[i];
+
+      // Negation is urnary and appears in the same token as the term it negates
+      // We don't consider a lone '-' as a negation. Lone operators are ignored because
+      // the user is probably about to type something to apply the operator to and we 
+      // don't want to prematurely hide any VMs
+      if (token.startsWith('-') && token.length > 1) {
+        const term = new SearchTerm(SearchOperator.Negate, [token.substring(1)]);
+        parsed.push(term);
+      } else if (token.length == 1) {
+        // This is a lone urnary operator - currently just means a lone '-' character
+        // ignore it so we don't discard matches that don't contain a literal '-' char
+        continue;
+      } else if (token.startsWith('\"')) {
+        // Exact match - find all tokens wrapped by quotes and consider them a single term
+        const [term, newIndex] = this.parseExactMatch(i, tokens);
+        i = newIndex;
+        parsed.push(term);
+      } else {
+        // This term has not been modified by an urnary operator but we still need to check for binary operators
+        // which is currently just OR - search has AND behavior by default, no need for an AND operator
+        
+        // Normal token
+        if (i >= tokens.length - 1 || !this.isBinOp(tokens[i + 1])) {
+          let term: SearchTerm;
+          // This token is escaped so discard the slash and search for a literal occurance of the token.
+          // Used to search for literal keyword/operators
+          if (this.isEscaped(tokens[i])) {
+            term = new SearchTerm(SearchOperator.None, [token.substr(1)]);
+          } else {
+            term = new SearchTerm(SearchOperator.None, [token]);
+          } 
+          parsed.push(term);
+        } else if (this.isBinOp(tokens[i + 1])) {
+          // A binary operator is being applied
+          const [term, newIndex] = this.parseBinaryOp(i, tokens);
+          i = newIndex;
+          parsed.push(term);
+        }
+      }
+    }
+    return parsed;
+  }
+
+  private isBinOp(tok: string): boolean {
+    const lower = tok.toLowerCase();
+    return lower == 'or';
+  }
+
+  private isEscaped(tok: string): boolean {
+    return tok.startsWith('\\');
+  }
+
+  /**
+   * Parse an expression with a binary operator
+   * @param i the current index into tokens
+   * @param tokens the list of tokens
+   */
+  private parseBinaryOp(i: number, tokens: string[]): [SearchTerm, number] {
+    const token = tokens[i];
+    const lower = tokens[i + 1].toLowerCase();
+    let term: SearchTerm;
+    if (lower == 'or') {
```

Comment:
could you use isBinOp() here?

Reply:
isBinOp() checks if a token is any binary operator, then I have to check which specific one it is. Since OR is currently the only binary operator, these checks are identical, but if we add more binary operators like xor or nand or something then we'll need the separate checks.

Reply:
okay that's fair, I just saw some duplicate code in there.  approved!